nengo_loihi/builder/decode_neurons.py

```diff
@@ -0,0 +1,311 @@
+import numpy as np
+
+import nengo
+
+from nengo_loihi.neurongroup import NeuronGroup
+from nengo_loihi.neurons import (
+    LoihiSpikingRectifiedLinear,
+    NIF,
+)
+from nengo_loihi.synapses import Synapses
+
+
+class DecodeNeurons(object):
+    """Defines parameters for a group of decode neurons.
+
+    DecodeNeurons are used on the chip to facilitate NEF-style connections,
+    where activities from a neural ensemble are first transformed into a
+    decoded value (which is stored in the activities and synapses of the
+    spiking decode neurons), before being passed on to another ensemble
+    (via that ensemble's encoders).
+
+    Parameters
+    ----------
+    dt : float
+        Time step used by the simulator.
+    """
+    def __init__(self, dt=0.001):
+        self.dt = dt
+
+    def __str__(self):
+        return "%s(dt=%0.3g)" % (type(self).__name__, self.dt)
+
+    def get_group(self, weights, group_label=None, syn_label=None):
+        """Get a NeuronGroup for implementing neurons on the chip.
+
+        Parameters
+        ----------
+        weights : (d, n) ndarray
+            Weights that project the ``n`` inputs to the ``d`` dimensions
+            represented by these neurons. Typically, the inputs will be neurons
+            belonging to an Ensemble, and these weights will be decoders.
+        group_label : string (Default: None)
+            Optional label for the NeuronGroup.
+        syn_label : string (Default: None)
+            Optional label for the Synapses.
+
+        Returns
+        -------
+        cx : CompartmentGroup
```

Comment:
Should be a `NeuronGroup` correct? Also since we're generally getting rid of the `cx` name it might be more clear to call this variable something else.